libbeat/common/kubernetes/util.go

```diff
@@ -158,6 +189,16 @@ func machineID() string {
 	return ""
 }
 
+// GetNamespace gets namespace from serviceaccount when beat is in cluster.
+func (hd *hostDiscovery) GetNamespace() (string, error) {
+	return InClusterNamespace()
```

Comment:
Maybe this function should be part of this `HostDiscovery`

Reply:
But the same function is used [in another place](https://sourcegraph.com/github.com/elastic/beats@77e83f975717f16fe21f1602e989e099f451763c/-/blob/x-pack/elastic-agent/pkg/composable/providers/kubernetesleaderelection/kubernetes_leaderelection.go?L71:24), so maybe is due to that.

I agree that `InClusterNamespace` looks a lot like an implementation detail of `hostDiscovery`.